### DISCUSSION TITLE
Mysterious Quarto error "Error: File `scrartcl.cls' not found" magically solved

### DISCUSSION THREAD

**Discussion by filipwastberg** [quarto-dev/quarto-cli#11663] [Q&A] [answered]
### Description

Hi! 

This is no longer an issue for us, but we'd like to understand why the following happened. We are running Posit Workbench with Quarto and TinyTex. We are running everything on a Red Hat Enterprise Linux 8.

TinyTex is installed system wide and is found in quarto check:

<details>
  <summary>quarto check</summary>
     ```

    quarto check
    Quarto 1.5.57
    [✓] Checking versions of quarto binary dependencies...
      Pandoc version 3.2.0: OK
      Dart Sass version 1.70.0: OK
      Deno version 1.41.0: OK
      Typst version 0.11.0: OK
    [✓] Checking versions of quarto dependencies......OK
    [✓] Checking Quarto installation......OK
      Version: 1.5.57
      Path: /usr/lib/rstudio-server/bin/quarto/bin
 
    [✓] Checking tools....................OK
      TinyTeX: (external install)
      Chromium: (not installed)
 
    [✓] Checking LaTeX....................OK
      Using: Installation From Path
      Path: /usr/local/bin
      Version: 2024
 
    [✓] Checking basic markdown render....OK
 
    [✓] Checking Python 3 installation....OK
      Version: 3.9.19
      Path: /tmp/RtmprVjl3P/rstudio/terminal/python3
      Jupyter: (None)
 
      Jupyter is not available in this Python installation.
      Install with python3 -m pip install jupyter
 
    [✓] Checking R installation...........OK
      Version: 4.4.1
      Path: /opt/R/4.4.1/lib/R
      LibPaths:
        - /home/user/R/x86_64-pc-linux-gnu-library/4.4
        - /opt/R/4.4.1/lib/R/library
      knitr: 1.49
      rmarkdown: 2.29
 
    [✓] Checking Knitr engine render......OK
    

</details>


We opened a Quarto document, just the default:

<details>
  <summary>quarto document</summary>

    ````qmd
    ---
    title: "Untitled"
    format: pdf
    editor: visual
    ---
    
    ## Quarto
    
    Quarto enables you to weave together content and executable code into a finished document. To learn more about Quarto see <https://quarto.org>.
    
    ## Running Code
    
    When you click the **Render** button a document will be generated that includes both content and the output of embedded code. You can embed code like this:
    
    ```{r}
    1 + 1
    ```
    
    You can add options to executable code like this
    
    ```{r}
    #| echo: false
    2 * 2
    ```
    
    The `echo: false` option disables the printing of code (only output is displayed).

    ```
</details>

But when we rendered it, we got a mysterious error from TinyTex:

```
Rendering PDFrunning xelatex - 1  This is XeTeX, Version 3.141592653-2.6-0.999996 (TeX Live 2024) (preloaded format=xelatex)
   restricted \write18 enabled.
  entering extended mode
  updating tlmgrupdating existing packagesfinding package for scrartcl.clsERROR: compilation failed- no matching packagesLaTeX Error: File `scrartcl.cls' not found.

Type X to quit or <RETURN> to proceed,
or enter new name. (Default extension: cls)

Enter file name: 
! Emergency stop.
<read *> 
         
l.10 ^^M
         

see test.log for more information.
```

We googled around, found [this old issue](https://github.com/quarto-dev/quarto-cli/issues/1281), restarted the session and tried to render again. This time it started the installation of something called koma-script:
```
Entering extended mode
  updating tlmgrupdating existing packagesfinding package for scrartcl.cls> 1 package to install> installing koma-script
```

But it failed and we got another error that we didn't save, but it said something in style with "Choose another mirror or wait a while/a day"

We restarted the session again and this time it managed to install.

```
Rendering PDFrunning xelatex - 1  This is XeTeX, Version 3.141592653-2.6-0.999996 (TeX Live 2024) (preloaded format=xelatex)
   restricted \write18 enabled.
  entering extended mode
  updating tlmgrupdating existing packagesfinding package for scrartcl.cls> 1 package to install> installing koma-script (1 of 1)  This is XeTeX, Version 3.141592653-2.6-0.999996 (TeX Live 2024) (preloaded format=xelatex)
   restricted \write18 enabled.
  entering extended mode
  finding package for caption.sty> 1 package to install> installing caption (1 of 1)  This is XeTeX, Version 3.141592653-2.6-0.999996 (TeX Live 2024) (preloaded format=xelatex)
   restricted \write18 enabled.
  entering extended mode
  running xelatex - 2  This is XeTeX, Version 3.141592653-2.6-0.999996 (TeX Live 2024) (preloaded format=xelatex)
   restricted \write18 enabled.
  entering extended mode
  

Output created: test.pdf
Watching files for changes
Browse at: p/56b65a36/web/viewer.html
```

Does anyone have any idea of what might be going on? We are happy that it works, but can't figure out why it didn't the first time?

**Comment by mcanouil** [comment]
Did you update or change your LaTeX distribution? (seems to be an issue unrelated to Quarto)

Did you update Quarto? (there were some changes over time on how Quarto + TinyTex try to be smart about ato-installing missing LaTeX tools/packages).

**Reply by filipwastberg** [reply]
No, this was pretty much a fresh install of Quarto and TinyTex, albeit not with 1.6. But I'm guessing that Quarto notices the TinyTex and then tries to be smart and auto-install stuff that might be specific for Quarto + TinyTex? I was just a bit suprised that it failed the first 3-4 times and then magically it managed to download. It might not be related to Quarto, but since the error occurred during `quarto render` I wanted to know if anyone had seen something similar or what is actually happening behind the scenes.

**Reply by mcanouil** [reply, marked as answer]
Since resources are online, it could have been some resource availability / network issues.

We can go on wild goose chase and make guesses but it won't help or change anything unless there is a clear way to reproduce.
